src/sinks/util/buffer/metrics/split.rs

```diff
@@ -0,0 +1,320 @@
+use std::collections::VecDeque;
+
+use vector_core::event::{metric::MetricData, Metric, MetricValue};
+
+#[allow(clippy::large_enum_variant)]
+enum SplitState {
+    Single(Option<Metric>),
+    Multiple(VecDeque<Metric>),
+}
+
+/// An iterator that returns the result of a metric split operation.
+pub struct SplitIterator {
+    state: SplitState,
+}
+
+impl SplitIterator {
+    /// Creates an iterator for a single metric.
+    pub const fn single(metric: Metric) -> Self {
+        Self {
+            state: SplitState::Single(Some(metric)),
+        }
+    }
+
+    /// Creates an iterator for multiple metrics.
+    pub fn multiple<I>(metrics: I) -> Self
+    where
+        I: Into<VecDeque<Metric>>,
+    {
+        Self {
+            state: SplitState::Multiple(metrics.into()),
+        }
+    }
+}
+
+impl Iterator for SplitIterator {
+    type Item = Metric;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match &mut self.state {
+            SplitState::Single(metric) => metric.take(),
+            SplitState::Multiple(metrics) => metrics.pop_front(),
+        }
+    }
+}
+
+/// Splits a metric into potentially multiple metrics.
+///
+/// In some cases, a single metric may represent multiple fundamental metrics: an aggregated summary or histogram can
+/// represent a count, sum, and subtotals for a given measurement. These metrics may be able to be handled
+/// natively/directly in a sink, but in other cases, those fundamental metrics may need to be extracted and operated on individually.
+///
+/// This trait defines a simple interface for defining custom rules about what metrics to split and when to split them.
+pub trait MetricSplit {
```

Comment:
Nothing is immediately planned, no.

I expressed the idea to Jesse that I would like to somehow collapse this, and `MetricNormalize`, into a trait that is more akin to `tower::Service` (sans any futures-related machinery, since we don't need it) so that we could build more middleware-oriented helpers, without necessarily being constrained to these specific traits.

That was a little too much to bite off just to solve the problem at hand, though, so I went with the trait-based approach to codify the _intent_ of treating splitting as a generic metric transformation that could be useful, in the same vein as metric normalization.... rather than just a very one-off, specific struct.

Maybe a little too far in the trait direction, but that's basically where my mind was at when writing this.